### PR TITLE
Added note about multiple Reacts, since this error commonly implies that.

### PR DESCRIPTION
**Pull request by jimfb**
Added note about multiple Reacts, since this error commonly implies that.

### CODE REVIEW

**Comment by jimfb** [comment]
@spicyj you have context on this one.

Lint failure is from a different file, unrelated to change.  Should be fixed independently.

**Comment by zpao** [comment]
(lint is some false positive thanks to build caching, cleared and retriggered - I hit it on master too)

**Comment by maoziliang** [comment]
How about add a magic token in the global object. Use the token to determine whether a past React had been  loaded. And then print a warning ?

**Comment by jimfb** [comment]
@maoziliang We tried that, but it ended up warning people in unit tests and we had to revert the diff.

@zpao or @spicyj ready to merge?

**Comment by jimfb** [comment]
Ping.

**Comment by jimfb** [comment]
@zpao or @spicyj Is there a reason we are not moving this forward, or should I keep pinging the thread?  Soon I'll get as good at this as pingbot!

**Comment by spicyj** [comment]
I want to criticize the too-long wording but haven't gotten around to it yet…

**Comment by jimfb** [comment]
@spicyj Comment overall or specifically my addition?

**Comment by spicyj** [comment]
Mostly it's just really long now.

**Comment by syranide** [comment]
@jimfb An idea would be to add a property (not attribute) to React root nodes with a per-React instance random value. Then for all the various asserts that can be affected by multiple Reacts we first check whether the property of the root node of that React tree has the value we expect, if it doesn't we know there's a different instance present.

**Comment by jimfb** [comment]
@spicyj Better?  It is now 4 lines instead of 7 lines :P.

@syranide That's not a bad idea.  If the root node is a web component, it might respond badly to having a random property set on it, but maybe we skip the check for web components?  It's worth looking into, but not worth blocking this fix.  This is a trivial mitigation that will catch the most common case.

File: src/renderers/shared/reconciler/ReactOwner.js [hunk 1]
```diff
@@ -70,11 +70,10 @@ var ReactOwner = {
...
       ReactOwner.isValidOwner(owner),
-      'addComponentAsRefTo(...): Only a ReactOwner can have refs. This ' +
-      'usually means that you\'re trying to add a ref to a component that ' +
-      'doesn\'t have an owner (that is, was not created inside of another ' +
-      'component\'s `render` method). Try rendering this component inside of ' +
-      'a new top-level component which will hold the ref.'
+      'addComponentAsRefTo(...): Only a ReactOwner can have refs. You might ' +
+      'be adding a ref to a component that was not created inside a component\'s ' +
+      '`render` method, or you have multiple copies of React loaded ' +
+      '(details: http://fb.me/react-refs-must-have-owner)'
```
Comment (spicyj) [inline src/renderers/shared/reconciler/ReactOwner.js]: https, period at the end of the sentence

**Comment by spicyj** [comment]
Okay, I guess this is fine. Can you shorten the remove message too? (Not sure I've ever seen it though since you'd always hit the add invariant first in practice.)

**Comment by jimfb** [comment]
@spicyj Yeah, is it even possible to hit that remove path?  Seems like you'd need to add the ref in order to remove it, so it would be impossible to hit.

**Pull request merged**